packages/interactivity/docs/2-api-reference.md

```diff
@@ -363,20 +365,90 @@ store( "myPlugin", {
 </details>
 <br/>
 
-The `wp-on` directive is executed each time the associated event is triggered. 
+≈
 
 The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.
 
-#### `wp-watch` 
+#### `wp-on-window`
+
+It allows to attach global window events like `resize`, `copy`, `focus` and then execute a defined callback when those happen.
+
+[List of supported window events.](https://developer.mozilla.org/en-US/docs/Web/API/Window#events)
+
+> The syntax of this directive is `data-wp-on--[window-event]` (like `data-wp-on-window--resize`
+or `data-wp-on-window--languagechange`).
+
+_Example of `wp-on-window` directive_
+
+```php
+<div data-wp-on-window--resize="callbacks.logWidth"></div>
+```
+
+<details>
+	<summary><em>See store used with the directive above</em></summary>
+
+```js
+store( "myPlugin", {
+  callbacks: {
+   logWidth() {
+			console.log( 'Window width: ', window.innerWidth );
+		},
+  },
+} );
+```
+
+</details>
+<br/>
+
+≈
+
+The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.
+
+#### `wp-on-document`
+
+It allows to attach global document events like `scroll`, `mousemove`, `keydown` and then execute a defined callback when those happen.
 
-It runs a callback **when the node is created and runs it again when the state or context changes**. 
+[List of supported document events.](https://developer.mozilla.org/en-US/docs/Web/API/Document#events)
+
+> The syntax of this directive is `data-wp-on-document--[document-event]` (like `data-wp-on-document--keydown`
+or `data-wp-on-document--languagechange`).
+
+_Example of `wp-on-document` directive_
+
+```php
+<div data-wp-on-document--keydown="callbacks.logKeydown"></div>
+```
+
+<details>
+	<summary><em>See store used with the directive above</em></summary>
+
+```js
+store( "myPlugin", {
+  callbacks: {
+   logKeydown(event) {
+			console.log( 'Key pressed: ', event.key );
+		},
+  },
+} );
+```
+
+</details>
+<br/>
+
+≈
+
+The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.
```

Comment:
```suggestion
The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.  When the element is removed from the DOM, the event listener is also removed.
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57931/commits/c8635c4132b548f97eb4136c39044881c8221867